test/e2e/service_options_test.go

```diff
@@ -163,6 +163,21 @@ func TestServiceOptions(t *testing.T) {
 	validateServiceInitScale(r, "svc11", "2")
 	t.Log("delete service")
 	test.ServiceDelete(r, "svc11")
+
```

Comment:
From the tests, what I'm missing a bit are the following cases:

* `--annotation` set the same annotation on *both*, service and revision
* `--annotation-revision` alone does *not* set the service annotation (the check is only that the revision annotation is set)
* Same negative check for service annotations is missing.

Also I would refactor the test to have a single validation method, that takes two maps for service and revision annotations respectively where each of the maps can be also nil it should not be set.

Reply:
otherwise, looks good to me so we are close to get the PR merged.

Reply:
I tried the negative tests with `len(annotation) == 0` and `annotation == nil`. Both fails. I think some annotations always get set automatically by default. So how do we test that no service annotations is set when we use --annotation-revision ?